lib/fileset/default.nix

```diff
@@ -156,7 +157,7 @@ If a directory does not recursively contain any file, it is omitted from the sto
           lib.fileset.toSource: `root` is of type ${typeOf root}, but it should be a path instead.''
     # Currently all Nix paths have the same filesystem root, but this could change in the future.
     # See also ../path/README.md
-    else if rootFilesystemRoot != filesetFilesystemRoot then
+    else if ! fileset._internalIsEmptySingleton && rootFilesystemRoot != filesetFilesystemRoot then
```

Comment:
This will print `fileset: root "null"` in case of error, which won't be very helpful if you don't know there is such a thing as an empty file set without a base path.

Reply:
Root is a path, it can never be null